src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/CodeWriter.cs

```diff
@@ -90,11 +90,83 @@ private static void InitUserSecrets(string projectPath)
             
             if (result.ExitCode != 0)
             {
+                if (!jsonOutput)
+                {
+                    Console.Write("FAILED\n");
+                }
                 throw new Exception("Error while running dotnet-user-secrets init");
             }
+            else
+            {
+                if (!jsonOutput)
+                {
+                    Console.Write("SUCCESS\n");
+                }
+            }
         }
 
-        private static void SetUserSecerets(string projectPath, string key, string value)
+        public static void AddPackage(string packageName, string packageVersion, string tfm, bool jsonOutput)
+        {
+            if (!string.IsNullOrEmpty(packageName) && ((!string.IsNullOrEmpty(packageVersion)) || (!string.IsNullOrEmpty(tfm))))
+            {
+                var errors = new List<string>();
+                var output = new List<string>();
+                var arguments = new List<string>();
+                arguments.Add("package");
+                arguments.Add(packageName);
+                if (!string.IsNullOrEmpty(packageVersion))
+                {
+                    arguments.Add("-v");
+                    arguments.Add(packageVersion);
+                }
+
+                if (IsTfmPreRelease(tfm))
+                {
+                    arguments.Add("--prerelease");
+                }
+
+                if (!string.IsNullOrEmpty(tfm))
+                {
+                    arguments.Add("-f");
+                    arguments.Add(tfm);
+                }
+                if (!jsonOutput)
+                {
+                    Console.Write($"\nAdding package {packageName} . . . ");
+                }
+
+                var result = Command.CreateDotNet(
+                    "add",
+                    arguments.ToArray())
+                    .OnErrorLine(e => errors.Add(e))
+                    .OnOutputLine(o => output.Add(o))
```

Comment:
Do we need to do anything with 'output'?

Reply:
Not particularly no. Package failure is more interesting which would be populated in errors. If the package is added successfully, we can perform a sanity check later. This is for the next update mostly.